internal/api_client/sddc_manager_client.go

```diff
@@ -234,17 +220,59 @@ func (sddcManagerClient *SddcManagerClient) getTask(ctx context.Context, taskId
 	}
 	// reset the counter
 	sddcManagerClient.getTaskRetries = 0
-	return getTaskResult.Payload, nil
+
+	return task, nil
 }
 
 func (sddcManagerClient *SddcManagerClient) retryTask(ctx context.Context, taskId string) error {
 	apiClient := sddcManagerClient.ApiClient
-	retryTaskParams := tasks.NewRetryTaskParamsWithTimeout(constants.DefaultVcfApiCallTimeout).
-		WithContext(ctx)
-	retryTaskParams.ID = taskId
-	_, err := apiClient.Tasks.RetryTask(retryTaskParams)
+	_, err := apiClient.RetryTaskWithResponse(ctx, taskId)
 	if err != nil {
 		return err
 	}
 	return nil
 }
+
+// GetResponseAs attempts to parse the response body into the provided type
+// If it fails it attempts to parse it as a vcf.Error.
+func GetResponseAs[T interface{}](body []byte, status int) (*T, *vcf.Error) {
```

Comment:
Why not pass the response object and get it's body and status code inside this function? If possible, this way the response code won't be mentioned in the files with the resources

Reply:
I have an idea about this but it would require a tweak in the SDK and I'd like to keep it for a second iteration of changes.

I can change the template for the models so that it adds a `func Body() []byte`.
Then we'd be able to define an interface that each model adheres to and we could use that as the parameter for `GetResponseAs`.  Then it would be as simple as `X.GetBody()` and `X.GetStatusCode()`

Reply:
Here it is https://github.com/vmware/vcf-sdk-go/pull/46

And this is what `GetResponseAs` would look like

```
type Response interface {
	GetBody() []byte
	StatusCode() int
}

func GetResponseAs[T interface{}](res Response) (*T, *vcf.Error) {
	if res.StatusCode() < 200 || res.StatusCode() >= 300 {
		return nil, GetError(res.GetBody())
	}

	var resp T
	if json.Unmarshal(res.GetBody(), &resp) != nil {
		return nil, nil
	}

	return &resp, nil
}
```

Reply:
Great! If you can fix it later, then ship it!